### PR TITLE
[TFLite] Add int8 and int16x8 support for CONV_3D operator

### PR DESCRIPTION
**Pull request by georgeedward2000**
Hello,

This PR adds quantizable int8 and int16x8 implementations of Conv3D operator. 
It also adds support for 5D tensors per axis (/per channel) quantization.

Thanks,
Eddie

### CODE REVIEW

**Comment by abattery** [comment]
@jianlijianli could you review this change?

**Comment by gbaned** [comment]
@jianlijianli  Can you please review this PR ? Thanks!

**Comment by gbaned** [comment]
@georgeedward2000 Can you please resolve conflicts? Thanks!

**Comment by gbaned** [comment]
Hi @jianlijianli Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
@georgeedward2000 Can you please resolve conflicts? Thank you!

**Comment by SaoirseARM** [comment]
> @georgeedward2000 Can you please resolve conflicts? Thank you!

Hi, 
Conflicts should be resolved now.
Thanks,
Saoirse

**Comment by gbaned** [comment]
@georgeedward2000 Can you please resolve conflicts? Thank you!

**Comment by abattery** [comment]
@alankelly could you review this PR request?

File: tensorflow/lite/kernels/internal/runtime_shape.h
Comment (alankelly) [inline tensorflow/lite/kernels/internal/runtime_shape.h:251]: This shouldn't be needed anymore once Offset has been removed from the loop above.

**Comment by gbaned** [comment]
@georgeedward2000  Can you please check @alankelly's comments and keep us posted ? Thank you!

File: tensorflow/lite/kernels/internal/reference/integer_ops/conv3d.h
Comment (Tessil) [reply tensorflow/lite/kernels/internal/reference/integer_ops/conv3d.h]: Yes, we used it in the same way as the float Conv3D reference kernel. I updated it, thanks!

File: tensorflow/lite/kernels/internal/runtime_shape.h
Comment (Tessil) [reply tensorflow/lite/kernels/internal/runtime_shape.h:251]: It is still used in `FillPerChannelMinMax` https://github.com/tensorflow/tensorflow/pull/51527/files#diff-26ec535332e9824d3089caee17d4c466a758ba7f5f3a5b4fa8a6ed11e3edd243R138

**Comment by Tessil** [comment]
@gbaned @alankelly Thanks, I updated the PR.

File: tensorflow/lite/kernels/internal/reference/integer_ops/conv3d.h
Comment (alankelly) [reply tensorflow/lite/kernels/internal/reference/integer_ops/conv3d.h]: Thanks!

File: tensorflow/compiler/mlir/lite/ir/tfl_ops.td
Comment (abattery) [inline tensorflow/compiler/mlir/lite/ir/tfl_ops.td:5369]: Apparently, there are more discussions for which the per channel dimension location should be supported in this operator in the long term.

Could you provide the rationale for your choice here for the future decision?

**Review by abattery — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/mlir/lite/ir/tfl_ops.td
Comment (Tessil) [reply tensorflow/compiler/mlir/lite/ir/tfl_ops.td:5369]: Sorry, I'm not sure to understand. We use per channel quantization for the Conv3D, so we use the output channel dimension (dimension 4) of the filters (operand 1). Though I might have misunderstood the `AffineOpCoefficient` usage.

**Comment by gbaned** [comment]
@georgeedward2000 Can you please check @abattery's comments and keep us posted ? Thank you!

File: tensorflow/compiler/mlir/lite/ir/tfl_ops.td
Comment (alankelly) [reply tensorflow/compiler/mlir/lite/ir/tfl_ops.td:5369]: @abattery Can you please respond/resolve this thread? Thanks

File: tensorflow/lite/tools/versioning/runtime_version.cc
Comment (alankelly) [inline tensorflow/lite/tools/versioning/runtime_version.cc:360]: 2.12.0

**Review by alankelly — COMMENTED** [review]
@arfaian for converter changes and @abattery for quantization changes. Can you please approve if OK for you? Thanks

Kernel changes LGTM

File: tensorflow/lite/tools/versioning/runtime_version.cc
Comment (Tessil) [reply tensorflow/lite/tools/versioning/runtime_version.cc:360]: Updated, thanks.

**Comment by gbaned** [comment]
Hi @abattery Can you please take a look on this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @abattery Can you please take a look on this PR ? Thank you!

**Review by alankelly — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @georgeedward2000 Can you please resolve conflicts? Thank you!

**Comment by Tessil** [comment]
@gbaned Thanks, I fixed the conflicts

@alankelly I dusted off a bit the PR as some problems appeared after the different merge conflicts resolutions, if you could review it when you have some time.

Thanks!

**Comment by gbaned** [comment]
Hi @alankelly Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @alankelly Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @alankelly Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @georgeedward2000 Can you please resolve conflicts? Thank you!

**Comment by SaoirseARM** [comment]
> Hi @georgeedward2000 Can you please resolve conflicts? Thank you!

Thanks! 
merge conflicts are resolved.
Best regards,

Saoirse

**Comment by keerthanakadiri** [comment]
Hi @ junjiang-lab , Can you please review this PR? Thank you !

**Comment by keerthanakadiri** [comment]
Hi @georgeedward2000 , Can you please resolve the conflicts? Thank you!

**Comment by keerthanakadiri** [comment]
Hi @georgeedward2000 , Can you please resolve the conflicts? Thank you!

**Comment by keerthanakadiri** [comment]
Hi @georgeedward2000 , Can you please resolve the conflicts? Thank you!

**Comment by keerthanakadiri** [comment]
Hi @georgeedward2000 , Can you please resolve the conflicts? Thank you!

**Comment by keerthanakadiri** [comment]
Hi @georgeedward2000 , Can you please resolve the conflicts? Thank you!

**Comment by keerthanakadiri** [comment]
Hi @georgeedward2000 , Can you please resolve the conflicts? Thank you!

**Comment by keerthanakadiri** [comment]
Hi @georgeedward2000 , Can you please resolve the conflicts? Thank you!

**Pull request closed without merging**
